lib/controller/transfercontroller.js

```diff
@@ -225,6 +226,31 @@ class TransferController extends EventEmitter {
             }
         });
     }
+
+    /**
+     * Informs the controller that the transfer for a given part has begun.
+     *
+     * @param {TransferPart} part Part that is now transferring.
+     */
+    transferringPart(part) {
+        const { transferAsset } = part;
+        if (!this.transferringAssets.has(transferAsset)) {
```

Comment:
Do we need to use the full object or could we use any kind of id?

Reply:
We could try using the URL of the asset. Is there a performance impact to using the full object?

Reply:
Not performance, I'm just wondering what having two equal transferAssets means. According to the documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#same-value-zero_equality), it would check references of the objects. Is this good enough here?

Reply:
I think you can track this in `TransferAsset` by storing the transfer start time in the object, it doesn't need to be tracked in the controller.